### ISSUE TITLE
Observable in message of custom validator

### ISSUE THREAD

**Issue by jonasvdb1** [ngx-formly/ngx-formly#3601]
In v5 the 'message' in a custom validator could return a message as an observable, which allowed to use the ngx-translate stream method:

```
validators: {
   ip: {
     expression: (c) => /(\d{1,3}\.){3}\d{1,3}/.test(c.value),
     message: (error, field: FormlyFieldConfig) => this.translate.stream('form.error.ip', {ip: field.formControl.value}),
   },
 },
```

In the new v6 it looks like it is not possible anymore to return the message as an observable.

```
interface FormlyValidation<T, R> {
    validation?: (string | T)[] | any;
    [key: string]: FormlyValidatorFn<R> | FormlyValidatorExpressionFn<R> | any;
}
declare type FormlyValidatorFn<T> = (control: AbstractControl, field: FormlyFieldConfig) => T;
interface FormlyValidatorExpressionFn<T> {
    expression: FormlyValidatorFn<T>;
    message: string | ((error: any, field: FormlyFieldConfig) => string);
}
```

Is there another way to pass an observable in the message for a custom validator?

**Comment by aitboudad** [comment]
Hi, we didn't change that part in `v6` so I think it can work as before, does it work is you set `message` property as `any` ?

**Comment by jonasvdb1** [comment]
Yes, everything is still working but in v5 it was clearer that the message could return an observable.
Can the Observable<string> type be added to the message property in `FormlyValidatorExpressionFn<T>` to make it clearer?

https://github.com/ngx-formly/ngx-formly/blob/3a3a8722c113132aa40e6c2de61402b2ec9dc396/src/core/src/lib/models/fieldconfig.ts#L278-L281

```
interface FormlyValidatorExpressionFn<T> {
  expression: FormlyValidatorFn<T>;
  message: string | ((error: any, field: FormlyFieldConfig) => string | Observable<string>);
}
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.1.2](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.2) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
